src/rabbit_amqqueue.erl

```diff
@@ -274,9 +274,15 @@ declare(QueueName, Durable, AutoDelete, Args, Owner, Node) ->
                                       recoverable_slaves = [],
                                       gm_pids            = [],
                                       state              = live})),
-    Node = rabbit_mirror_queue_misc:initial_queue_node(Q, Node),
+
+    Node1 = case rabbit_queue_master_location_misc:get_location(Q)  of
+              {ok, Node0}  -> Node0;
+              {error, _}   -> Node
+            end,
+
+    Node1 = rabbit_mirror_queue_misc:initial_queue_node(Q, Node1),
```

Comment:
@michaelklishin please have a look at our conversations on 17/07/2015 and 20/07/2015 on my understanding of this. For the changes you're commenting on, @videlalvaro also already had a look at them. Just the test code which hadn't gone through PR.